### PR TITLE
Rolling PR for flight-scheduler

### PR DESCRIPTION
**Pull request by WilliamMcCumstie**
This PR will be used to track the progress of `flight-scheduler`

### CODE REVIEW

File: builders/flight-scheduler/opt/flight/etc/service/types/scheduler-controller/start.sh [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+log_file="${flight_ROOT}"/var/log/scheduler-controller/falcon.log
+
+# The wrapper script which redirects falcon's logs and allows logrotate to
+# reopen the file descriptors. The first argument must be the log path, all
+# subsequent arguments are directly passed to falcon
+tool_bg "${flight_ROOT}"/bin/ruby \
+  "${flight_ROOT}"/opt/scheduler/controller/bin/start \
+  "$log_file" --threaded -n 2 -b "$addr"
+
+# Wait up to 10ish seconds for falcon to start
```
Comment (benarmston) [inline builders/flight-scheduler/opt/flight/etc/service/types/scheduler-controller/start.sh:52]: We have a new script to get the falcon pid in https://github.com/openflighthpc/flight-scheduler-controller/pull/16.

**Review by benarmston — CHANGES_REQUESTED** [review]
@WilliamMcCumstie can you explain why the stderr redirection/reopening is required.  Why would a simple `2> /whatever/path.log` not be sufficient?

File: builders/flight-scheduler/opt/flight/etc/service/types/scheduler-controller/start.sh [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+# Set the address, log path, var dir, and pid file path
+addr=http://127.0.0.1:918
+log_file="${flight_ROOT}"/var/log/scheduler-controller/falcon.log
+
+# The wrapper script which redirects falcon's logs and allows logrotate to
+# reopen the file descriptors. The first argument must be the log path, all
+# subsequent arguments are directly passed to falcon
+tool_bg "${flight_ROOT}"/bin/ruby \
+  "${flight_ROOT}"/opt/scheduler/controller/bin/start \
+  "$log_file" --threaded -n 2 -b "$addr"
```
[the remark this answers is not in the document]
Comment (WilliamMcCumstie) [reply builders/flight-scheduler/opt/flight/etc/service/types/scheduler-controller/start.sh]: Updated to use `falcon-host`

File: builders/flight-scheduler/opt/flight/etc/service/types/scheduler-controller/start.sh [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+# Required to support initializers.
+export USER=$(whoami)
+# Required to correctly handle output parsing.
+if [ -f /etc/locale.conf ]; then
+  . /etc/locale.conf
+fi
+export LANG=${LANG:-en_US.UTF-8}
+
+# Move to the source code dirctory
+cd "$flight_ROOT"/opt/scheduler/controller
```
[the remark this answers is not in the document]
Comment (WilliamMcCumstie) [reply builders/flight-scheduler/opt/flight/etc/service/types/scheduler-controller/start.sh]: Appears not

File: builders/flight-scheduler/opt/flight/etc/service/types/scheduler-controller/start.sh (hunk 1 above)
Comment (WilliamMcCumstie) [reply builders/flight-scheduler/opt/flight/etc/service/types/scheduler-controller/start.sh]: Switched

**Comment by WilliamMcCumstie** [comment]
> @WilliamMcCumstie can you explain why the stderr redirection/reopening is required. Why would a simple `2> /whatever/path.log` not be sufficient?

Updated to use `copytruncate` instead of reopening the logs

**Comment by benarmston** [comment]
LGTM.

**Pull request merged**